src/sources/models.js

```diff
@@ -178,6 +182,13 @@ class Resource {
   get baseName() {
     return this.baseParts.join("_");
   }
+  versionInfo() {
+    /**
+     * Interrogates the resource index to find the appropriate version of the
+     * resource (which includes all subresources appropriate for this resource)
+     */
+    throw new Error("versionInfo(versionDescriptor) must be implemented by Resource subclasses");
+  }
```

Comment:
Suggestion: Make defining `versionInfo()` optional by replacing this placeholder with the [default warning](https://github.com/nextstrain/nextstrain.org/blob/354a458fcc7fa65c89e40fe1142cc8e92a76777c/src/sources/community.js#L194-L199). Then it would only need to be defined where it's actually used, for now just `CoreDataset`.

Reply:
This paired with my previous suggestion directly above https://github.com/nextstrain/nextstrain.org/pull/719#discussion_r1376886961 would mean no code changes necessary in community/fetch/groups.

Reply:
I've made this the default behaviour by allowing the method to work on the base class, and requiring subclasses to overwrite it as needed. This gets rid of the code changes to community / fetch / groups.